src/Transform.jl

```diff
@@ -0,0 +1,374 @@
+# Convert between different parameterizations.
+
+module Transform
+
+using Celeste
+using CelesteTypes
+
+import Util
+
+export rect_transform, free_transform, identity_transform, DataTransform
+
+#export unconstrain_vp, rect_unconstrain_vp, constrain_vp, rect_constrain_vp
+#export unconstrain_vp!, rect_unconstrain_vp!, constrain_vp!, rect_constrain_vp!
+#export rect_unconstrain_sensitive_float, unconstrain_sensitive_float
+
+type DataTransform
+	# Functiones to move between a ModelParameters object and a
+	# transformation of the data for optimization.
+    #
+    # to_vp: A function that takes transformed parameters and returns variational parameters
+    # from_vp: A function that takes variational parameters and returned transformed parameters
+    # to_vp!: A function that takes (transformed paramters, variational parameters) and updates
+    #   the variational parameters in place
+    # from_vp!: A function that takes (variational paramters, transformed parameters) and updates
+    #   the transformed parameters in place
+    # ... 
+    # transform_sensitive_float: A function that takes (sensitive float, model parameters)
+    #   where the sensitive float contains partial derivatives with respect to the
+    #   variational parameters and returns a sensitive float with total derivatives with
+    #   respect to the transformed parameters.
+
+	to_vp::Function
+	from_vp::Function
+	to_vp!::Function
+	from_vp!::Function
+    vp_to_vector::Function
+    vector_to_vp!::Function
```

Comment:
Yeah.  I have to figure out what's going on in my text editor.  Without going into details it's behaving kind of strangely here.  (It's Sublime, which I'm currently on the fence about purchasing.  How this resolves itself may affect my decision.)